domains/certificates/Query.API/API/Query.API/Projections/CertificatesByOwnerProjection.cs

```diff
@@ -1,119 +1,186 @@
+
+        foreach (var @event in events)
         {
-            view.Certificates[@event.CertificateId] = new CertificateView
+            switch (@event)
             {
-                CertificateId = @event.CertificateId,
-                DateFrom = @event.Period.DateFrom,
-                DateTo = @event.Period.DateTo,
-                Quantity = @event.ShieldedQuantity.Value,
-                GSRN = @event.ShieldedGSRN.Value,
-                GridArea = @event.GridArea,
-                TechCode = @event.Technology.TechCode,
-                FuelCode = @event.Technology.FuelCode,
-                Status = CertificateStatus.Issued // Transfer is only allowed if certificate is issued
-            };
+                case ProductionCertificateCreated productionCertificateCreated:
+                    CreateCertificatesByOwnerView(operations,
+                        productionCertificateCreated);
+                    break;
+                case ProductionCertificateIssued productionCertificateIssued:
+                    ApplyProductionCertificateIssued(operations,
+                        productionCertificateIssued);
+                    break;
+                case ProductionCertificateRejected productionCertificateRejected:
+                    ApplyProductionCertificateRejected(
+                        operations, productionCertificateRejected);
+                    break;
+                case ProductionCertificateTransferred productionCertificateTransferred:
+                    ApplyProductionCertificateTransferred(operations, productionCertificateTransferred);
+                    break;
+                default:
+                    throw new InvalidOperationException();
+            }
         }
+
+        operations.Store(views.ToArray());
+    }
+
+    private static void ApplyProductionCertificateRejected(IDocumentOperations operations,
+        ProductionCertificateRejected productionCertificateRejected)
+    {
+        var view = views
+                       .FirstOrDefault(it =>
+                           it.Certificates.ContainsKey(productionCertificateRejected.CertificateId)) ??
+                   GetCertificatesByOwnerView(operations, productionCertificateRejected.CertificateId);
+
+        view.Certificates[productionCertificateRejected.CertificateId].Status = CertificateStatus.Rejected;
+        views.Add(view);
+    }
+
+    private static void ApplyProductionCertificateIssued(IDocumentOperations operations,
+        ProductionCertificateIssued productionCertificateIssued)
+    {
+        var view = views
+                       .FirstOrDefault(it =>
+                           it.Certificates.ContainsKey(productionCertificateIssued.CertificateId)) ??
+                   GetCertificatesByOwnerView(operations, productionCertificateIssued.CertificateId);
+
+        view.Certificates[productionCertificateIssued.CertificateId].Status = CertificateStatus.Issued;
+        views.Add(view);
```

Comment:
Please see the comment for ApplyProductionCertificateRejected about adding to `views`